src/platform/Tizen/BLEManagerImpl.cpp

```diff
@@ -37,35 +40,694 @@ namespace Internal {
+
+        ChipLogProgress(DeviceLayer, "GATT Service Registered");
+        mFlags.Set(Flags::kAppRegistered);
+        ExitNow();
+    }
+
+    if (mServiceMode == ConnectivityManager::kCHIPoBLEServiceMode_Enabled && mFlags.Has(Flags::kAdvertisingEnabled))
+    {
+        if (!mFlags.Has(Flags::kAdvertising))
+        {
+            ret = StartAdvertising();
+            VerifyOrExit(ret == BT_ERROR_NONE, ChipLogError(DeviceLayer, "Start Advertising Failed. ret: %d", ret));
+        }
+        else if (mFlags.Has(Flags::kAdvertisingRefreshNeeded))
+        {
+            ChipLogProgress(DeviceLayer, "Advertising Refreshed Needed. Stop Advertising");
+            ret = StopAdvertising();
+            VerifyOrExit(ret == BT_ERROR_NONE, ChipLogError(DeviceLayer, "Stop Advertising Failed. ret: %d", ret));
+        }
+    }
+    else if (mFlags.Has(Flags::kAdvertising))
+    {
+        ChipLogProgress(DeviceLayer, "Stop Advertising");
+
+        ret = StopAdvertising();
+        VerifyOrExit(ret == BT_ERROR_NONE, ChipLogError(DeviceLayer, "Stop Advertising Failed. ret: %d", ret));
+
+        ret = bt_adapter_le_destroy_advertiser(mAdvertiser);
+        VerifyOrExit(ret == BT_ERROR_NONE, ChipLogError(DeviceLayer, "bt_adapter_le_destroy_advertiser() failed. ret: %d", ret));
+        mAdvertiser = NULL;
+    }
+
+exit:
+    if (ret != BT_ERROR_NONE)
+    {
+        mServiceMode = ConnectivityManager::kCHIPoBLEServiceMode_Disabled;
+    }
+}
+
+void BLEManagerImpl::DriveBLEState(intptr_t arg)
+{
+    sInstance.DriveBLEState();
+}
+
+CHIP_ERROR BLEManagerImpl::_Init(void)
+{
+    CHIP_ERROR err;
+    bool ret;
+
+    err = BleLayer::Init(this, this, this, &DeviceLayer::SystemLayer());
+    SuccessOrExit(err);
+
+    mServiceMode = ConnectivityManager::kCHIPoBLEServiceMode_Enabled;
+
+    ChipLogProgress(DeviceLayer, "Initialize BLE");
+    ret = MainLoop::Instance().Init(_BleInitialize);
+    VerifyOrExit(ret != false, err = CHIP_ERROR_INTERNAL);
+
+    PlatformMgr().ScheduleWork(DriveBLEState, 0);
+
+exit:
     return err;
 }
 
 CHIP_ERROR BLEManagerImpl::_SetCHIPoBLEServiceMode(CHIPoBLEServiceMode val)
 {
-    return CHIP_ERROR_NOT_IMPLEMENTED;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    VerifyOrExit(val != ConnectivityManager::kCHIPoBLEServiceMode_NotSupported, err = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(mServiceMode == ConnectivityManager::kCHIPoBLEServiceMode_NotSupported, err = CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE);
+
+    if (val != mServiceMode)
+    {
+        mServiceMode = val;
+        PlatformMgr().ScheduleWork(DriveBLEState, 0);
+    }
+
+exit:
+    return err;
 }
 
 CHIP_ERROR BLEManagerImpl::_SetAdvertisingEnabled(bool val)
 {
-    return CHIP_ERROR_NOT_IMPLEMENTED;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    if (mFlags.Has(Flags::kAdvertisingEnabled) != val)
+    {
+        mFlags.Set(Flags::kAdvertisingEnabled, val);
```

Comment:
What is the purpose of this if block?  Is the ScheduleWork call meant to be inside it too?  If not, why not just set the flag unconditionally?

Reply:
Thank you for your comment. I removed if block.